src/Service/Messenger.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace App\Service;
+
+
+use App\Model\Message;
+
+class Messenger
+{
+    /**
+     * @var SenderInterface[]
+     */
+    protected $senders = [];
+
+    /**
+     * Messenger constructor.
+     * @param SenderInterface[] $senders
+     */
+    public function __construct(array $senders = [])
```

Comment:
This method is fine, but I'd suggest a different approach: use foreach loop to go through the senders; check their types to make sure all those come from the right interface. Otherwise, you can create additional addSender() method so that you can use propper typehint other parameter.